### ISSUE TITLE
Group titles do not get passed to accessible announcements

### ISSUE THREAD

**Issue by Watso196** [JedWatson/react-select#6047]
When Groups are used to group options in the Select, these groupings are not announced to screen reader users. This is because the announcements screen reader users hear are controlled by the LiveRegion component, and no information about an option's Group is passed to that LiveRegion.

For example, when using the arrow keys to "focus" an option in a group, we would expect to hear something like "[option name], option, not selected, [group name], group" from screen readers. Instead, we don't hear any information about the group name or group role. Currently, there is no good way for React Select users to customize the LiveRegion content to change this read out unless we completely overwrite the LiveRegion content using ariaLiveMessages.

React Select should pass on Grouping information to the LiveRegion component so that it can be announced to screen reader users. Here is a codesandbox instance with an example of grouped options: https://codesandbox.io/p/devbox/react-select-v5-sandbox-forked-9zwtgv
